### PR TITLE
feat: add sharding for the :users scope

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

File: lib/realtime/syn_shards.ex [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          else: :syn.group_names(shard)
+
+      group_names ++ acc
+    end)
+    |> Enum.uniq()
+  end
+
+  @spec join(atom(), term(), pid()) :: :ok | {:error, term()}
+  def join(scope, name, pid) do
+    scope
```
Comment (chasers) [inline lib/realtime/syn_shards.ex:31]: Shard by pid instead? Otherwise a tenant will be pinned to one syn process.

Comment (abc3) [reply lib/realtime/syn_shards.ex:31]: Damn, such a stupid mistake. Of course, sharing by name doesn’t make sense in our case 🤦‍♂️🤦‍♂️🤦‍♂️

**Review by filipecabaco — APPROVED** [review]
this almost feels like it should be part of `:syn` itself 😅

**Pull request merged**